encodings/alp/src/array.rs

```diff
@@ -95,13 +95,14 @@ impl ALPArray {
 
     pub fn patches(&self) -> Option<Array> {
         self.metadata().patches_dtype.as_ref().map(|dt| {
-            self.as_ref().child(1, dt, self.len()).unwrap_or_else(|| {
-                vortex_panic!(
-                    "Missing patches with present metadata flag; patches dtype: {}, patches_len: {}",
+            self.as_ref().child(1, dt, self.len()).vortex_expect(
+                format!(
```

Comment:
don't do this, because it will greedily evaluate the string interpolation (whereas the previous one would do so lazily)